error: specialization of 'template<class IsPy37> const OwnedObject greenlet::Greenlet::context(typename IsPy37::IsIt) const' in different namespace

I'm seeing [this error in our autobuilders](http://autobuild.buildroot.net/results/eb6/eb67fe65c6ff245204c5c163b6704b1e5aa29a6a/build-end.log) for greenlet 2.0.1.
```cpp
creating build/temp.linux-x86_64-3.11/src/greenlet
/usr/bin/gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -O2 -I/home/buildroot/autobuild/instance-0/output-1/host/include -O2 -I/home/buildroot/autobuild/instance-0/output-1/host/include -I/home/buildroot/autobuild/instance-0/output-1/host/include -fPIC -I/home/buildroot/autobuild/instance-0/output-1/host/include/python3.11 -c src/greenlet/greenlet.cpp -o build/temp.linux-x86_64-3.11/src/greenlet/greenlet.o
src/greenlet/greenlet.cpp:2511:64: error: specialization of 'template<class IsPy37> const OwnedObject greenlet::Greenlet::context(typename IsPy37::IsIt) const' in different namespace [-fpermissive]
 Greenlet::context<GREENLET_WHEN_PY37>(GREENLET_WHEN_PY37::Yes) const
                                                                ^~~~~
In file included from src/greenlet/greenlet_internal.hpp:20:0,
                 from src/greenlet/greenlet.cpp:19:
src/greenlet/greenlet_greenlet.hpp:333:27: error:   from definition of 'template<class IsPy37> const OwnedObject greenlet::Greenlet::context(typename IsPy37::IsIt) const' [-fpermissive]
         const OwnedObject context(const typename IsPy37::IsIt=nullptr) const;
                           ^~~~~~~
src/greenlet/greenlet.cpp:2540:71: error: specialization of 'template<class IsPy37> const OwnedObject greenlet::Greenlet::context(typename IsPy37::IsIt) const' in different namespace [-fpermissive]
 Greenlet::context<GREENLET_WHEN_NOT_PY37>(GREENLET_WHEN_NOT_PY37::No) const
                                                                       ^~~~~
In file included from src/greenlet/greenlet_internal.hpp:20:0,
                 from src/greenlet/greenlet.cpp:19:
src/greenlet/greenlet_greenlet.hpp:333:27: error:   from definition of 'template<class IsPy37> const OwnedObject greenlet::Greenlet::context(typename IsPy37::IsIt) const' [-fpermissive]
         const OwnedObject context(const typename IsPy37::IsIt=nullptr) const;
                           ^~~~~~~
src/greenlet/greenlet.cpp:2562:89: error: specialization of 'template<class IsPy37> void greenlet::Greenlet::context(greenlet::refs::BorrowedObject, typename IsPy37::IsIt)' in different namespace [-fpermissive]
 void Greenlet::context<GREENLET_WHEN_PY37>(BorrowedObject given, GREENLET_WHEN_PY37::Yes)
                                                                                         ^
In file included from src/greenlet/greenlet_internal.hpp:20:0,
                 from src/greenlet/greenlet.cpp:19:
src/greenlet/greenlet_greenlet.hpp:336:21: error:   from definition of 'template<class IsPy37> void greenlet::Greenlet::context(greenlet::refs::BorrowedObject, typename IsPy37::IsIt)' [-fpermissive]
         inline void context(refs::BorrowedObject new_context, typename IsPy37::IsIt=nullptr);
                     ^~~~~~~
src/greenlet/greenlet.cpp:2597:99: error: specialization of 'template<class IsPy37> void greenlet::Greenlet::context(greenlet::refs::BorrowedObject, typename IsPy37::IsIt)' in different namespace [-fpermissive]
 Greenlet::context<GREENLET_WHEN_NOT_PY37>(BorrowedObject UNUSED(given), GREENLET_WHEN_NOT_PY37::No)
                                                                                                   ^
In file included from src/greenlet/greenlet_internal.hpp:20:0,
                 from src/greenlet/greenlet.cpp:19:
src/greenlet/greenlet_greenlet.hpp:336:21: error:   from definition of 'template<class IsPy37> void greenlet::Greenlet::context(greenlet::refs::BorrowedObject, typename IsPy37::IsIt)' [-fpermissive]
         inline void context(refs::BorrowedObject new_context, typename IsPy37::IsIt=nullptr);
                     ^~~~~~~
error: command '/usr/bin/gcc' failed with exit code 1
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -9,7 +9,7 @@
   was currently active. See `issue 332
   <https://github.com/python-greenlet/greenlet/issues/332>`_.
 - Various compilation and standards conformance fixes. See #335, #336,
-  #300, #302.
+  #300, #302, #334.
 
 
 

--- a/src/greenlet/greenlet.cpp
+++ b/src/greenlet/greenlet.cpp
@@ -2506,6 +2506,9 @@ green_setparent(BorrowedGreenlet self, BorrowedObject nparent, void* UNUSED(cont
 #    define GREENLET_NO_CONTEXTVARS_REASON "This Python interpreter"
 #endif
 
+namespace greenlet
+{
+
 template<>
 const OwnedObject
 Greenlet::context<GREENLET_WHEN_PY37>(GREENLET_WHEN_PY37::Yes) const
@@ -2543,19 +2546,6 @@ Greenlet::context<GREENLET_WHEN_NOT_PY37>(GREENLET_WHEN_NOT_PY37::No) const
                          GREENLET_NO_CONTEXTVARS_REASON
                          "does not support context variables"
     );
-}
-
-static PyObject*
-green_getcontext(const PyGreenlet* self, void* UNUSED(context))
-{
-    const Greenlet *const g = self->pimpl;
-    try {
-        OwnedObject result(g->context<G_IS_PY37>());
-        return result.relinquish_ownership();
-    }
-    catch(const PyErrOccurred&) {
-        return nullptr;
-    }
 }
 
 template<>
@@ -2600,6 +2590,21 @@ Greenlet::context<GREENLET_WHEN_NOT_PY37>(BorrowedObject UNUSED(given), GREENLET
                          GREENLET_NO_CONTEXTVARS_REASON
                          "does not support context variables"
     );
+}
+
+};
+
+static PyObject*
+green_getcontext(const PyGreenlet* self, void* UNUSED(context))
+{
+    const Greenlet *const g = self->pimpl;
+    try {
+        OwnedObject result(g->context<G_IS_PY37>());
+        return result.relinquish_ownership();
+    }
+    catch(const PyErrOccurred&) {
+        return nullptr;
+    }
 }
 
 static int

